packages/graph/hash_graph/lib/graph/src/api/rest/entity_type.rs

```diff
@@ -170,15 +170,15 @@ async fn create_entity_type<P: StorePool + Send>(
 )]
 async fn get_entity_types_by_query<P: StorePool + Send>(
     pool: Extension<Arc<P>>,
-    Json(query): Json<StructuralQuery>,
+    Json(query): Json<serde_json::Value>,
 ) -> Result<Json<Subgraph>, StatusCode> {
     pool.acquire()
         .map_err(|error| {
             tracing::error!(?error, "Could not acquire access to the store");
             StatusCode::INTERNAL_SERVER_ERROR
         })
         .and_then(|store| async move {
-            let mut query = NewStructuralQuery::try_from(query).map_err(|error| {
+            let mut query = StructuralQuery::deserialize(&query).map_err(|error| {
```

Comment:
Nice addition. We talked offline about this, and having explicit deserialize here means that Axum won't swallow the JSON errors from the extractor